### PR TITLE
DEVX-2258: Improve page titles of all client programming language exa…

### PR DESCRIPTION
**Pull request by ybyzek**
…mples

### Description 

https://confluentinc.atlassian.net/browse/DEVX-2258

_What behavior does this PR change, and why?_

Improve title pages to be more descriptive

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

- [x] Documentation

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

- [x] Documentation

### CODE REVIEW

**Review by awalther28 — APPROVED** [review]
Built docs locally and spot reviewed the client pages. I did look over all of the files changes, everything looks good to me. I appreciate the common title structure.

File: clients/docs/c.rst [hunk 1]
```diff
@@ -1,8 +1,7 @@
 .. _client-examples-c:
 
-C
-=
-
+Apache Kafka Producer and Consumer Example: C (librdkafka)
```
Comment (JimGalasyn) [inline clients/docs/c.rst]: You might want to use the `|ak|` token in the heading -- this actually works!

File: clients/docs/confluent-cli.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .. _client-examples-confluent-cli:
 
-Confluent CLI
-=============
+Apach Kafka Producer and Consumer Example: Confluent CLI
```
Comment (JimGalasyn) [inline clients/docs/confluent-cli.rst]:
```suggestion
Apache Kafka Producer and Consumer Example: Confluent CLI
```

File: clients/docs/confluent-cli.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
+========================================================
```
Comment (JimGalasyn) [inline clients/docs/confluent-cli.rst]:
```suggestion
=========================================================
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a couple of suggestions.

File: clients/docs/confluent-cli.rst (hunk 2 above)
Comment (ybyzek) [reply clients/docs/confluent-cli.rst]: And this is why tokens are great, thank you.

**Review by rmoff — APPROVED** [review]
LGTM 👍

**Pull request merged**